cases/util/fetcher.js

```diff
@@ -0,0 +1,30 @@
+import { fetch } from "./fetchShim";
+
+export const loggableFetch = async (requestURL, requestDictionary) => {
+  
+    const response = await fetch(requestURL,requestDictionary);
+    const json = await response.json();
+
+    const responseStatus = response.status;
+
+    const rawHeaders = response.headers.raw();
+    const formattedHeaders = Object.keys(rawHeaders).map(key => `${key}: ${rawHeaders[key]}`).join("\n")
+  
+    const requestMethod = requestDictionary["method"] ? requestDictionary["method"] : "GET";
+  
+    const logs = `
+    \n -----------------------------
+    \n⬇️ REQUEST ⬇️
+    \nURL: ${requestURL}
+    \nMETHOD: ${requestMethod}
+    \nHEADERS: \n${JSON.stringify(requestDictionary["headers"], null, 2)}
+    \nBODY: \n${JSON.stringify(requestDictionary["body"], null, 2)}
+    
+    \n⬇️ RESPONSE ⬇️
+    \nSTATUS: ${response.status}
+    \nHEADERS: \n${formattedHeaders}
+    \nBODY: \n${JSON.stringify(json, null, 2)}
+    \n -----------------------------`;
+    
+    return [json,  responseStatus, logs];
```

Comment:
can you do this as an object instead of an array? `return {json, responseStatus, logs}

Reply:
or `return {json, status: responseStatus, logs}`.  javascript has a shorthand for objects where if you don't do the key:value, you can put in the same object as its own name.

`{json}` is equivalent to `{json: json}`